src/dosext/dpmi/dnative.c

```diff
@@ -59,6 +59,10 @@ static void copy_context(sigcontext_t *d, sigcontext_t *s)
 
 static void copy_to_dpmi(sigcontext_t *scp, cpuctx_t *s)
 {
+#ifdef __x86_64__
+  /* needs to clear high part of rip or sighandler stops giving err codes :( */
+  scp->rip = 0;
```

Comment:
Hi @amluto I am hitting something very
strange here. I've found that, when returning
to compatibility mode of x86_64 via sigreturn(),
you need to clear the high dword of a RIP
register. If you don't do that, the sighandler
no longer sets the valid error codes for GPF,
even though the trapno is set properly.
Do you have any idea what's going on here?
I thought high parts of 64bit regs are completely
irrelevant in compatibility mode, so why should
I clear RIP?